boat-scaffold/src/main/java/com/backbase/oss/codegen/angular/BoatAngularGenerator.java

```diff
@@ -157,7 +148,13 @@ public void processOpts() {
         addSupportingFiles();
 
         processOpt(NG_VERSION,
-            value -> applyAngularVersion(new SemVer(value)),
+            value -> {
+                if (value.contains("||")) {
```

Comment:
Is this conditional really adding anything? Won't passing the string directly to `applyAngularVersion` have the same result?

Reply:
It's in place to fork for a specific set of logic, related to unions

Reply:
I understand that, but I think the end result will be the same if you pass any previously valid value into the new overload.

e.g. calling`applyAngularVersion("10.0.0")` (passing the string) will calculate the `greatestVersion` as `10.0.0` and then set the `NG_VERSION` and call `addDependencies` accordingly. Which is exactly the same result as calling `applyAngularVersion(new SemVer("10.0.0")`.

(except for the >= 10 check I mention in my other comment)